src/order/antisymmetrization.lean

```diff
@@ -0,0 +1,175 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import order.hom.basic
+
+/-!
+# Turning a preorder into a partial order
+
+This file allows to make a preorder into a partial order by quotienting out the elements `a`, `b`
+such that `a ≤ b` and `b ≤ a`.
+
+`antisymmetrization` is a functor from `Preorder` to `PartialOrder`. See `Preorder_to_PartialOrder`.
+
+## Main declarations
+
+* `le_equiv`: The antisymmetrization relation. `le_equiv a b` means that `a` and `b` are less than
+  each other.
+* `antisymmetrization α`: The quotient of `α` by `le_equiv`. Even when `α` is just a preorder,
+  `antisymmetrization α` is a partial order.
+-/
+
+open order_dual
+
+variables {α β : Type*}
+
+section has_le
+variables [has_le α]
+
+/-- The antisymmetrization relation. -/
+def le_equiv (a b : α) : Prop := a ≤ b ∧ b ≤ a
```

Comment:
I don't like this name. Any ideas?

It corresponds to `category_theory.zag`, so maybe `le_zag`?

Reply:
Somewhat long, but what about `comparable`?

Reply:
Wait no, it is neither `zag` nor `comparable`, that would an `or` not an `and`.

Reply:
Ahrghy, you're totally right of course.

Reply:
Any reason not to use `le_and_le`?

Reply:
Renamed it to `antisymm_rel`